runtime/server/metrics_views.go

```diff
@@ -78,89 +58,223 @@ func (s *Server) MetricsViewToplist(ctx context.Context, req *api.MetricsViewTop
+	}
+
+	return resp, nil
+}
+
+// MetricsViewTotals implements RuntimeService
+func (s *Server) MetricsViewTotals(ctx context.Context, req *api.MetricsViewTotalsRequest) (*api.MetricsViewTotalsResponse, error) {
+	sql, args, err := buildMetricsTotalsSql(req)
 	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+		return nil, status.Errorf(codes.Internal, "error building query: %s", err.Error())
 	}
 
-	data, err := rowsToData(rows)
+	meta, data, err := s.runQuery(ctx, req.InstanceId, sql, args)
 	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+		return nil, err
 	}
 
-	resp := &api.MetricsViewTimeSeriesResponse{
+	resp := &api.MetricsViewTotalsResponse{
 		Meta: meta,
-		Data: data,
+		Data: data[0],
 	}
 
 	return resp, nil
 }
 
-// MetricsViewTotals implements RuntimeService
-func (s *Server) MetricsViewTotals(ctx context.Context, req *api.MetricsViewTotalsRequest) (*api.MetricsViewTotalsResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method not implemented")
+func (s *Server) runQuery(ctx context.Context, instanceId string, sql string, args []any) ([]*api.SchemaColumn, []*structpb.Struct, error) {
+	rows, err := s.query(ctx, instanceId, &drivers.Statement{
+		Query: sql,
+		Args:  args,
+	})
+	if err != nil {
+		return nil, nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	defer rows.Close()
+
+	meta, err := rowsToMeta(rows)
+	if err != nil {
+		return nil, nil, status.Error(codes.Internal, err.Error())
+	}
+
+	data, err := rowsToData(rows)
+	if err != nil {
+		return nil, nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return meta, data, nil
 }
 
 func buildMetricsTimeSeriesSQL(req *api.MetricsViewTimeSeriesRequest) (string, []any, error) {
-	timeField := "time"
-	timeCol := fmt.Sprintf("DATE_TRUNC(%s, %s) AS %s", timeField, req.TimeGranularity, timeField)
+	// TODO: get from Catalog
+	timeField := "timestamp"
+	timeCol := fmt.Sprintf("DATE_TRUNC('%s', %s) AS %s", req.TimeGranularity, timeField, timeField)
 	selectCols := append([]string{timeCol}, req.MeasureNames...)
 
-	whereClause := "time >= ? AND time < ? "
-	args := []any{time.UnixMilli(req.TimeStart), time.UnixMilli(req.TimeEnd)}
+	whereClause := fmt.Sprintf("%s >= epoch_ms(?) AND %s < epoch_ms(?) ", timeField, timeField)
+	args := []any{req.TimeStart, req.TimeEnd}
 
-	if req.Filter != nil && req.Filter.Include != nil {
-		clause, clauseArgs, err := buildFilterClause(req.Filter.Include, "IN")
+	if req.Filter != nil {
+		clause, clauseArgs, err := buildFilterClauseForMetricsViewFilter(req.Filter)
 		if err != nil {
 			return "", nil, err
 		}
 		whereClause += clause
 		args = append(args, clauseArgs...)
 	}
 
-	if req.Filter != nil && req.Filter.Exclude != nil {
-		clause, clauseArgs, err := buildFilterClause(req.Filter.Exclude, "NOT IN")
+	sql := fmt.Sprintf("SELECT %s FROM %s WHERE %s GROUP BY %s LIMIT 1000",
+		strings.Join(selectCols, ", "), req.MetricsViewName, whereClause, timeField)
```

Comment:
I don't have a strong preference here, so if you prefer, don't change this, but often when I wrap long function calls in Go, I tend to put each arg on a new line, like this:

```go
// One line
sql := fmt.Sprintf("SELECT %s FROM %s WHERE %s GROUP BY %s LIMIT 1000", strings.Join(selectCols, ", "), req.MetricsViewName, whereClause, timeField)

// Wrapped line
sql := fmt.Sprintf(
	"SELECT %s FROM %s WHERE %s GROUP BY %s LIMIT 1000",
	strings.Join(selectCols, ", "),
	req.MetricsViewName,
	whereClause,
	timeField,
)
```

Reply:
The 2nd is better, i believe auto formatting does this. I couldnt get auto formatting working in this repo. I will spend some time and fix this in the file.